### PR TITLE
[Fix] SSA aliasing in StableHLOToBatchNormTrainingOpConversionPattern

### PR DESCRIPTION
**Pull request by saiarthiraguram**
Github issue: [Link](https://github.com/tenstorrent/tt-mlir/issues/8561)

### Bug

`torch.nn.LayerNorm` on tt-xla silently produced `output_std ≈ input_std²` instead of normalized output. Invisible to unit-variance Gaussian tests (`1² = 1`) but catastrophic for transformer activations whose stddev is in the tens-to-thousands.

**Verified affected:** MapTR Nano R18 110e (end-to-end PCC = 0.0519 pre-fix).

**Scope by mechanism (not empirically tested beyond MapTR):**
- Bug fires for `torch.nn.LayerNorm` via the `torch_xla` eager dispatch path (the path used by `tests/runner/test_models.py`). PCC stays high for a single LayerNorm because magnitudes scale uniformly; failure becomes visible when multiple LayerNorms stack or the inflated values flow into ops that overflow.
- Does not fire for `torch.compile(backend="tt")` (Dynamo decomposes `LayerNorm` into primitive ops before StableHLO emission).
- Does not fire for `torch.nn.BatchNorm1d/2d` callers (they pass real, distinct weight/bias parameters; no CSE aliasing).

## Root cause

PyTorch/XLA lowers `nn.LayerNorm` to `aten::native_batch_norm` → `stablehlo.batch_norm_training` with synthetic-ones scale / synthetic-zeros offset. `ttnn.batch_norm_training` requires `running_mean` / `running_var` operands (declared `[MemWrite]`), but `stablehlo.batch_norm_training` doesn't supply them — the conversion synthesized them as `ttir.zeros` and `ttir.ones`. After lowering, CSE merged these with the user-side scale=ones / offset=zeros constants (identical values), so the same SSA value ended up in both a `[MemWrite]` running-stats slot and the read-only weight/bias slot. tt-metal's `batch_norm` in training mode then clobbered weight/bias while updating running stats in place, producing `output = input × input_std`.

## Fix

Replace the synthetic `ttir.zeros` / `ttir.ones` running stats with `ttir.empty`. `ttir::EmptyOp` declares a `MemoryEffects::Allocate` effect that CSE respects, so it stays distinct from the scale/offset constants. Safe because momentum=1.0 makes the running-stats update independent of the initial value, and the running stats aren't returned to the XLA caller.

### Isolated LayerNorm sanity

**Before:**

[layernorm_sanity_BEFORE_FIX.log](https://github.com/user-attachments/files/27985574/layernorm_sanity_BEFORE_FIX.log)

**After:**

[layernorm_sanity_AFTER_FIX.log](https://github.com/user-attachments/files/27985568/layernorm_sanity_AFTER_FIX.log)

### MapTR Nano R18 110e end-to-end pytest

```
 pytest -svv tests/runner/test_models.py::test_all_models_torch[maptr/pytorch-Nano_R18_110e-single_device-inference]
```

**Before** (Attach maxpool PR):

[maptr_full_BEFORE_FIX.log](https://github.com/user-attachments/files/27985530/maptr_full_BEFORE_FIX.log)

**After** :

[maptr_full_AFTER_FIX.log](https://github.com/user-attachments/files/27985518/maptr_full_AFTER_FIX.log)

MapTR end-to-end PCC: 0.0519 → 0.4978. Per-stage outputs (img_backbone, img_neck, transformer, pts_bbox_head) match CPU at PCC = 0.9999 — the model math is correct. The remaining 0.4978 → 0.99 gap is a top-k tie-break sensitivity in `pts_bbox_head.get_bboxes()` post-processing.

### CODE REVIEW

**Comment by saiarthiraguram** [comment]
The requirement for this PR is not required as of now, will reopen if required
